Applications/Spire/Include/Spire/Ui/NavigationView.hpp

```diff
@@ -0,0 +1,121 @@
+#ifndef SPIRE_NAVIGATION_VIEW_HPP
+#define SPIRE_NAVIGATION_VIEW_HPP
+#include <any>
+#include <QStackedWidget>
+#include <QWidget>
+#include "Spire/Spire/AssociativeValueModel.hpp"
+#include "Spire/Spire/ValueModel.hpp"
+#include "Spire/Ui/Ui.hpp"
+
+namespace Spire {
+
+  /**
+   * Represents a component which holds the NavigationList and associated page
+   * components.
+   */
+  class NavigationView : public QWidget {
+    public:
+
+      /**
+       * A ValueModel over an integer to represent the current of the
+       * NavigationView.
+       */
+      using CurrentModel = ValueModel<int>;
+
+      /**
+       * A LocalValueModel over an integer to represent the local
+       * current of the NavigationView.
+       */
+      using LocalCurrentModel = LocalValueModel<int>;
+
+      /**
+       * Constructs a NavigationView using a LocalCurrentModel.
+       * @param parent The parent widget.
+       */
+      explicit NavigationView(QWidget* parent = nullptr);
+
+      /**
+       * Constructs a NavigationView.
+       * @param current_model The current value's model.
+       * @param parent The parent widget.
+       */
+      explicit NavigationView(std::shared_ptr<CurrentModel> current_model,
+        QWidget* parent = nullptr);
+
+      /** Returns the current model. */
+      const std::shared_ptr<CurrentModel>& get_current_model() const;
+
+      /**
+       * Appends a tab with a page and label.
+       * @param page The component displayed in the page area.
+       * @param tab_label The text value associated with the page.
+       */
+      void add_tab(QWidget& page, const QString& tab_label);
```

Comment:
Reduce the use of adjectives that can be immediately inferred from context.

```tab_label``` -> ```label```